test/api/upload_test.go

```diff
@@ -158,3 +165,152 @@ func testUploadInfo(t *testing.T, at *test.AccountsTester) {
 		t.Fatalf("Expected 200OK and %d uploads, got %d and %d uploads.", 5, sc, len(ups))
 	}
 }
+
+// TestUploadedSkylinks ensures UploadsByPeriod returns the correct uploads.
+// This one relies on a clean DB, so it can't be run with the same tester as
+// other tests which create uploads.
+func TestUploadedSkylinks(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	dbName := test.DBNameForTest(t.Name())
+	at, err := test.NewAccountsTester(dbName, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		if errClose := at.Close(); errClose != nil {
+			t.Error(errors.AddContext(errClose, "failed to close account tester"))
+		}
+	}()
+	// Add a test user.
+	email := types.Email(t.Name() + "@example.com")
```

Comment:
Follow Up: might be a nice test helper to have a `CreateTestUser` that returns the needed inputs like email and password.